server.py

```diff
@@ -38,14 +38,32 @@ def index():
 def list_products():
     category        = request.args.get('category')
     discontinued    = str2bool(request.args.get('discontinued'))
-    results     = []
+    min_Price       = request.args.get('minPrice')
+    max_Price       = request.args.get('maxPrice')
+    price           = request.args.get('price')
+    limit           = request.args.get('limit')
+    results         = []
+    count           = 0
+    if (limit is not None and int(limit) < 0):
```

Comment:
As was discussed in: Issue https://github.com/NYU-DevOps-Charlie-CSCI-GA-3033-013/products/issues/15, ideally there is a default cut-off/limit value in place that can be overriden by the API user using the `limit` URL attribute.

Ideally, there would have been a separate pull request for this feature for Issue https://github.com/NYU-DevOps-Charlie-CSCI-GA-3033-013/products/issues/15, but it's fine we're addressing it now, especially since it touches the same code.

So I would refactor it in the following manner:
1. Inside the `list_products()` method, define a `limit` variable with a default cutoff value, e.g. 10.
2. Also inside the `list_products()` method, check for a (valid) `limit` URL attribute, (`request.args.get('limit')`); if there is one, then override the default with the value that the user gave. 
3. In the scenario where a user passes an invalid `limit` variable, I leave it to your discretion whether the API should simply ignore the value and use the default - or reply with an HTTP 404 response.